packages/fxa-settings/src/pages/ResetPassword/AccountRecoveryConfirmKey/index.tsx

```diff
@@ -74,46 +76,56 @@ const AccountRecoveryConfirmKey = ({
         AuthUiErrors.INVALID_RECOVERY_KEY
       );
       setErrorMessage(localizedErrorMessage);
-      setIsSubmitting(false);
     }
   };
 
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    if (errorMessage) {
+      setErrorMessage('');
+    }
+    // strip out any characters other than letters, numbers or spaces
+    // spaces are left in for ease of user review but removed on submit
+    const filteredCode = e.target.value.replace(/[^a-zA-Z0-9\s]/g, '');
+    e.target.value = filteredCode;
+    e.target.value.replace(/\s/g, '').length === 32
```

Comment:
Why the second `replace`?  The spaces were remove in the first `replace` call.  nit: `filteredCode` seems unnecessary; could just directly assign the return value of teh first `replace` into `e.target.value`.

Reply:
I'll see if I can make this clearer - the first `replace` filters out anything other than alphanumeric or spaces, but the second removes the spaces to check if the code has a valid format. Agree on the second point though.

Reply:
Oh haha yeah forgot about the `^`.  But why not replace the spaces also?

Reply:
Sorry I feel like I was unintentionally asking a simple question in a roundabout fashion... so if we want to check if there are 32 alphanumeric characters in the input, then why not directly go there?

Reply:
The intent was for the user to be able to see the spaces if they're copying the code (easier for them to review) but ignore them when checking whether or not to enable submission. Certainly easier to do away with that and just strip out the spaces. I'll update that way and ask Eduardo to take a look at the storybook.

Reply:
> The intent was for the user to be able to see the spaces if they're copying the code (easier for them to review) but ignore them when checking whether or not to enable submission.

Ah, gotcha.  That make sense.  Thank you for the explanation.